onnxscript/test/operator_test.py

```diff
@@ -42,6 +42,48 @@ def implicit_plus1(A: FLOAT["N"]) -> FLOAT["N"]:  # noqa: F821
 
         self.assertSame(explicit_plus1, implicit_plus1)
 
+    def test_bool_ops_binary(self):
+        @script(default_opset=op)
+        def py_two_or(a: BOOL, b: BOOL) -> BOOL:
+            return a or b
+
+        @script()
+        def onnx_two_or(a: BOOL, b: BOOL) -> BOOL:
+            return op.Or(a, b)
+
+        self.assertSame(py_two_or, onnx_two_or)
+
+        @script(default_opset=op)
+        def py_two_and(a: BOOL, b: BOOL) -> BOOL:
+            return a and b
+
+        @script()
+        def onnx_two_and(a: BOOL, b: BOOL) -> BOOL:
+            return op.And(a, b)
+
+        self.assertSame(py_two_and, onnx_two_and)
+
+    def test_bool_ops_many(self):
+        @script(default_opset=op)
+        def py_three_or(a: BOOL, b: BOOL, c: BOOL) -> BOOL:
+            return a or b or c
+
+        @script()
+        def onnx_three_or(a: BOOL, b: BOOL, c: BOOL) -> BOOL:
+            return op.Or(op.Or(a, b), c)
+
+        self.assertSame(py_three_or, onnx_three_or)
+
+        @script(default_opset=op)
+        def py_four_and(a: BOOL, b: BOOL, c: BOOL, d: BOOL) -> BOOL:
+            return a and b and c and d
```

Comment:
Do we also want the interleave and/or? just wondering.

Reply:
That shouldn't be necessary since the Python parser will create separate nodes. You can't have an `ast.BoolOp` with multiple operators.

For instance:

```python
a and b and c or d or e or f and g and h and i
```

Results in:

```python
BoolOp(
  op=Or(),
  values=[
    BoolOp(
      op=And(),
      values=['a', 'b', 'c']),
    'd',
    'e',
    BoolOp(
      op=And(),
      values=['f', 'g', 'h', 'i'])])
```